iphone/Maps/Classes/CustomViews/MapViewControls/TrafficButton/MWMTrafficButtonViewController.mm

```diff
@@ -36,6 +36,7 @@ @interface MWMTrafficButtonViewController () <MWMMapOverlayManagerObserver, Them
 @property(nonatomic) NSLayoutConstraint *topOffset;
 @property(nonatomic) NSLayoutConstraint *leftOffset;
 @property(nonatomic) CGRect availableArea;
+@property(nonatomic) MWMToast *toast;
```

Comment:
I'm not sure about this approach... to store toast on every controller that we have in the project (where it needed of cours). And the other issue can happen: sometimes we can have 2-3 toasts at the same time (not in the current place, but in the PlacePage) and when we call `hide()` manually we will close only the last toast stored in the property.
Maybe it would be better to incapsulate toast logic inside of Toast class and store toasts in the static property?
With this approach we can hide all toasts from every place we want.
Some kine of:
```swift
class Toast {
// ...
static private func toasts: [Toast] = []
static func hideAll() {
  toasts.forEach { $0.hide }
}
func show(...) {
  Toast.appent(toast)
  // ... showing code ... 
}

func hide(...) {
  //... hiding code ...
  Toast.remove(toast)
}

```

@biodranik what do you think?

Reply:
In the long term, it would be great if users could read all toasts without overlapping each other if they appear simultaneously (e.g. stack them, or show one after another). A simpler way to control toast visibility is also great.

How does Apple solve toast issue?

Reply:
Apple doesn't have toasts at all. Only push notifications.
This is why toasts are so different from an app to app.

Storing all toasts logic inside the Toast class helps us in the future to handle the toasts queue properly with animations and gesture recognizers.

Reply:
@v-lozko is it hard to implement it in a way described by @kirylkaveryn ? What do you think about that approach?

Reply:
I can give it a go

Reply:
@biodranik , @kirylkaveryn  I made the changes as suggested. Seems to work correctly

https://github.com/organicmaps/organicmaps/assets/156805389/e08717e5-3406-49a8-bc18-aa59e9646f2e